code/modules/mob/living/carbon/human/shock.dm

```diff
@@ -0,0 +1,74 @@
+/mob/living/carbon/human/var/traumatic_shock = 0
+/mob/living/carbon/human/var/shock_stage = 0
+
+// proc to find out in how much pain the mob is at the moment
+/mob/living/carbon/human/proc/updateshock()
+	traumatic_shock = getOxyLoss() + getToxLoss() + getFireLoss() + getBruteLoss() + getCloneLoss()
+
+	// broken or ripped off organs will add quite a bit of pain
+	for(var/thing in bodyparts)
+		var/obj/item/organ/external/BP = thing
+		if(BP.status & ORGAN_BROKEN && !(BP.status & ORGAN_SPLINTED) || BP.open)
+			traumatic_shock += 15
+
+	if(reagents)
+		for(var/datum/reagent/R in reagents.reagent_list)
+			if(R.shock_reduction)
+				traumatic_shock = max(0, traumatic_shock - R.shock_reduction) // now you too can varedit cyanide to reduce shock by 1000 - Iamgoofball
+	if(drunk)
```

Comment:
checking for drunk instead of `slurring` because I'm fairly certain that's what's intended. I don't think slurring magically reduces your pain level.